src/Microsoft.AspNet.SignalR.Client/Connection.cs

```diff
@@ -975,6 +976,26 @@ private static string CreateQueryString(IDictionary<string, string> queryString)
             return String.Join("&", queryString.Select(kvp => kvp.Key + "=" + kvp.Value).ToArray());
         }
 
+        /// <summary>
+        /// Gets the query from a url and removes the query string from the url passed
+        /// </summary>
+        /// <param name="url"></param>
+        /// <returns>the query string from the url passed and removes the query string from the url.
+        /// If a query string is not in the url it returns null and leaves the url untouched</returns>
+        private static string GetQueryStringFromUrl(ref string url)
+        {
+            if (url == null) throw new ArgumentNullException("url");
+            string queryString = null;
+            if (url.Contains("?"))
+            {
+                var urlSplitBits = url.Split('?');
+
+                queryString = queryString = urlSplitBits[1];
```

Comment:
This would not work if there was another question mark in the query string - I think the best way is to find the index of the first '?' and if it is not -1 return a substring starting after this `?`